quickfixj-core/src/main/java/quickfix/DefaultSessionFactory.java

```diff
@@ -131,9 +147,13 @@ public Session create(SessionID sessionID, SessionSettings settings) throws Conf
                         .getBool(sessionID, Session.SETTING_USE_DATA_DICTIONARY);
             }
 
-            DefaultDataDictionaryProvider dataDictionaryProvider = null;
+            EditableDataDictionaryProvider dataDictionaryProvider = null;
             if (useDataDictionary) {
-                dataDictionaryProvider = new DefaultDataDictionaryProvider();
+                if(editableDataDictionaryProvider == null) {
+                    dataDictionaryProvider = new DefaultDataDictionaryProvider();
+                } else {
+                    dataDictionaryProvider = editableDataDictionaryProvider;
```

Comment:
You now have a dataDictionaryProvider per SessionFactory rather than per Session. Do we need to think about this?

Perhaps we should be passing in a DataDictionaryProviderFactory.